dotCMS/src/main/java/com/dotcms/publisher/business/PublishAuditAPIImpl.java

```diff
@@ -380,4 +364,57 @@ public boolean isPublishRetry(final String bundleId) {
         return isRetry;
     }
 
+    @CloseDBIfOpened
+	@Override
+	public List<String> getBundleIdByStatus(final List<Status> statusList, final int limit, final int offset) throws DotDataException {
+		List<String> bundleIds = new ArrayList<>();
+		final DotConnect dc = new DotConnect();
+		String sql = SELECT_BUNDLEID_BY_STATUS;
+		if(statusList.size() > 1){
+			for(int i = 1;i<statusList.size();i++){
+				sql += OR_STATUS_CLAUSE;
+			}
+		}
+		dc.setSQL(sql);
+		dc.setMaxRows(limit);
+		dc.setStartRow(offset);
+		for(final Status status : statusList){
+			dc.addParam(status.getCode());
+		}
+
+		for(final Map<String,Object> resultMap : dc.loadObjectResults()) {
+			bundleIds.add((String) resultMap.get("bundle_id"));
+		}
+
+		return bundleIds;
+	}
+
+	@CloseDBIfOpened
+	@Override
+	public List<String> getBundleIdByStatusFilterByOwner(final List<Status> statusList,
+			final int limit, final int offset, final String userId) throws DotDataException {
+		List<String> bundleIds = new ArrayList<>();
+		final DotConnect dc = new DotConnect();
+		String sql = SELECT_BUNDLEID_BY_STATUS_AND_OWNER;
+		if(statusList.size() > 1){
+			for(int i = 1;i<statusList.size();i++){
+				sql += OR_STATUS_CLAUSE;
+			}
+		}
+		sql += ")";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Prefer StringBuilder (non-synchronized) or StringBuffer (synchronized) over += for concatenating strings](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4502981)